File: src/main/scala/us/zoom/data/dfence/providers/snowflake/models/opaque/ObjectTypeAlias.scala
package us.zoom.data.dfence.providers.snowflake.models.opaque

import us.zoom.data.dfence.providers.snowflake.grant.builder.SnowflakeObjectType

opaque type ObjectTypeAlias = String

object ObjectTypeAlias:
  def apply(snowflakeObjectType: SnowflakeObjectType): ObjectTypeAlias =
    snowflakeObjectType.getAliasFor()

File: src/main/java/us/zoom/data/dfence/providers/snowflake/revoke/models/wrappers/ObjectTypeAlias.java
package us.zoom.data.dfence.providers.snowflake.revoke.models.wrappers;

import us.zoom.data.dfence.providers.snowflake.grant.builder.SnowflakeObjectType;

public record ObjectTypeAlias(String value) {
  public static ObjectTypeAlias apply(SnowflakeObjectType snowflakeObjectType) {
    return new ObjectTypeAlias(snowflakeObjectType.getAliasFor());
  }
}
